src/http/client.rs

```diff
@@ -65,7 +65,7 @@ impl DefaultHttpClient {
 
   pub async fn execute(&self, request: reqwest::Request) -> reqwest_middleware::Result<Response> {
     log::info!("{} {} ", request.method(), request.url());
-    let response = self.client.execute(request).await?;
+    let response = self.client.execute(request).await?.error_for_status()?;
```

Comment:
Need a `http spec`  for this change.

Reply:
I've added the http_spec for this but the problem is that http_spec has mocked http client that implements that functionality by its logic and therefore I have to add error handler in http mock as well.

we may require full e2e testing to add proper tests

Reply:
added more sane test in server_spec